portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFolderServiceImpl.java

```diff
@@ -775,11 +776,24 @@ public DLFolder updateFolder(
 			int restrictionType, ServiceContext serviceContext)
 		throws PortalException {
 
-		ModelResourcePermissionUtil.check(
+		ModelResourcePermission<DLFolder> modelResourcePermission =
 			ModelResourcePermissionRegistryUtil.
-				<DLFolder>getModelResourcePermission(DLFolder.class.getName()),
-			getPermissionChecker(), serviceContext.getScopeGroupId(), folderId,
-			ActionKeys.UPDATE);
+				<DLFolder>getModelResourcePermission(DLFolder.class.getName());
+		PermissionChecker permissionChecker = getPermissionChecker();
+
+		if (!ModelResourcePermissionUtil.contains(
+				modelResourcePermission, permissionChecker,
+				serviceContext.getScopeGroupId(), folderId,
+				ActionKeys.ADVANCE_UPDATE) &&
+			!ModelResourcePermissionUtil.contains(
+				modelResourcePermission, permissionChecker,
+				serviceContext.getScopeGroupId(), folderId,
+				ActionKeys.UPDATE)) {
+
+			throw new PrincipalException.MustHavePermission(
```

Comment:
I am not completely sure if this models the or use cause, that is to say, the user needs ADVANCE_UPDATE permission or the other one.